### ISSUE TITLE
"an input to unknown location has changed" with `experiment-configuration-cache`

### ISSUE THREAD

**Issue by SimonMarquis** [gradle/develocity-build-validation-scripts#776]
While upgrading to the new Develocity naming, the [experiment-config-cache](https://github.com/gradle/develocity-build-validation-scripts/blob/main/.github/actions/gradle/experiment-config-cache/action.yml) workflow started acting weird.

Specifically, the 2 Gradle run (in the same GHA job) does not re-use the configuration cache, because of "Calculating task graph as configuration cache cannot be reused because an input to unknown location has changed." as you can see here: https://github.com/SimonMarquis/SealedObjectInstances/actions/runs/12432422305/job/34712511059#step:4:259

And the corresponding Gradle Scan does not show what it is but "an input to unknown location has changed	": https://scans.gradle.com/s/46lwbdtmths4c/performance/configuration#summary-configuration-caching-cache-invalidation-reason

On my local environment it works as expected. The configuration cache is first stored, then restored on 2nd run.

Any tips on how to debug this further?

EDIT: `--info` [does not seem to be helpful](https://github.com/SimonMarquis/SealedObjectInstances/actions/runs/12432846941/job/34713141858?pr=313#step:4:3518).
EDIT: using Gradle 8.11.1

**Comment by SimonMarquis** [comment]
I found 3 interesting things from the [Gradle configuration cache reports](https://github.com/SimonMarquis/SealedObjectInstances/actions/runs/12433108231?pr=313#artifacts):
- `GIT_TRACE_PERFORMANCE` input (env variable) is in the 1st execution only
- `durian.plugins.com.diffplug.common.base.Errors.Plugins.Log` input (system property) is also in the 1st execution
- there are 40 files from `.gradle/caches/modules-2/` that are also part of the 1st execution only.

**Comment by erichaagdev** [comment]
Hi, @SimonMarquis. Thanks for taking the time to report this. 

I can't think of a reason why the rename would cause this. My first theory is it has something to do with the [setup-gradle](https://github.com/gradle/develocity-build-validation-scripts/blob/4321ced6172ee23f6ed78a10f3103432fbd0ba8b/.github/actions/gradle/experiment-config-cache/action.yml#L15-L18) action that we include, though I haven't been able to investigate deeply. If you don't mind experimenting, I wonder if you could try to copy [the action](https://github.com/gradle/develocity-build-validation-scripts/blob/main/.github/actions/gradle/experiment-config-cache/action.yml#L15-L18) and exclude the setup-gradle step to see if it helps at all. Please let me know.

**Comment by SimonMarquis** [comment]
Unfortunately I got the same result: "Calculating task graph as configuration cache cannot be reused because an input to unknown location has changed."

https://github.com/SimonMarquis/SealedObjectInstances/actions/runs/12433656005?pr=313

**Comment by SimonMarquis** [comment]
It seems like that configuring Gradle with my own setup-gradle solved the issue. 🎉 
I tried to also force the [`cache-disabled: true`](https://github.com/SimonMarquis/SealedObjectInstances/pull/313/commits/c5e8adde8d31fea8ce3d4fc4d7e8f9711fd74b65) but it worked the same way 🤷

**Comment by SimonMarquis** [comment]
It seems like to only appear when running on the main (default) branch 🤔

**Comment by SimonMarquis** [comment]
Clearing GitHub action's cache fixes the issue, for 1 build...
Then it fails again 🤯

**Comment by erichaagdev** [comment]
Hi @SimonMarquis. Is this still an issue? I see your latest runs are successful.

**Comment by SimonMarquis** [comment]
I'm not 100% sure why/how it happens since Gradle does not really say what the input change is. The attempts only show flakiness when clearing the entire cache.
I suspected the issue was coming from spotless that I recently upgraded to a beta version, and I ended up reverting it.
I'll see when the stable version is released if the issue persists.

**Comment by SimonMarquis** [comment]
Indeed, this is still happening now with the same [Cache miss reason: an input to unknown location has changed](https://scans.gradle.com/s/npvwa57mphgzy/performance/configuration#summary-configuration-caching-cache-invalidation-reason).

**Comment by erichaagdev** [comment]
@SimonMarquis I don't believe this is an issue with the scripts or the experiment. We eliminated the possibility that it was caused by the rename by copying & pasting the action code. 

I see this issue https://github.com/gradle/gradle/issues/25469 that seems to be related to the error you're seeing. I would ask you to give it a 👍, but I see you already have. Since you seem to be able to reproduce it, perhaps you could share the steps to do so over in that issue? 

I am going to go ahead and close this issue since we determined this is not an issue caused by this project.
